### PR TITLE
Use ccache everywhere

### PR DESCRIPTION
**Pull request by vchuravy**
I installed ccache on Windows and the mac bot. 
While deleting files on Windows will still be slow, this allows us 
more consistent builds (and builds for release-0.6)

### CODE REVIEW

File: master/package.py [hunk 1]
```diff
@@ -19,42 +19,13 @@
     steps.Git(
         name="Julia checkout",
         repourl=util.Property('repository', default='git://github.com/JuliaLang/julia.git'),
-        mode='incremental',
-        method='clean',
+        mode='full',
+        method='fresh',
```
Comment (ararslan) [inline master/package.py]: Will this make submitting nuke jobs unnecessary?

Comment (staticfloat) [reply master/package.py]: Yes, hopefully.

**Comment by staticfloat** [comment]
Okay, I like this, but I agree with Tony, let's hold off on the `USE_BINARYBUILDER_LLVM=1`; let's keep that restricted to Travis CI stuff for now.

**Comment by vchuravy** [comment]
Rebased out the superflous commit.

**Comment by staticfloat** [comment]
Fingers crossed.  Let's see if this works.

**Pull request merged**